docs/contributing/code-contributions.md

```diff
@@ -0,0 +1,56 @@
+a pull request.
+
+## Making changes to the starter library
+
+Note: You don't need to follow these steps to submit to the starter library. This is only necessary if you'd like to contribute to the functionality of the starter library. To submit a starter, [follow these steps instead](/contributing/submit-to-starter-library/).
+
+To develop on the starter library, you'll need to supply a GitHub personal access token.
+
+1. Create a personal access token in your GitHub [Developer settings](https://github.com/settings/tokens).
+2. In the new token's settings, grant that token the "public_repo" scope.
+3. Create a file in the root of `www` called `.env.development`, and add the token to that file like so:
+
+```text:title=.env.development
+GITHUB_API_TOKEN=YOUR_TOKEN_HERE
+```
+
+The `.env.development` file is ignored by git. Your token should never be committed.
+
+## Contributing to the blog
+
+Note: Before adding a blog post ensure you have approval from a member of the Gatsby team. You can do this by [opening an issue](https://github.com/gatsbyjs/gatsby/issues/new/choose) or contacting [@gatsbyjs on Twitter](https://twitter.com/gatsbyjs).
+
+To add a new blog post to the gatsbyjs.org blog:
+
+- Clone [the Gatsby repo](https://github.com/gatsbyjs/gatsby/) and navigate to `/www`
+- Run `yarn` to install all of the website's dependencies.
+- Run `npm run develop` to preview the blog at `http://localhost:8000/blog`.
+- The content for the blog lives in the `/docs/blog` folder. Make additions or modifications here.
+- Add your avatar image to `/docs/blog/avatars`
+- Add your name to `/docs/blog/author.yaml`
+- Add a new folder following the pattern `/docs/blog/yyyy-mm-dd-title`. Within this newly created folder add an `index.md` file.
+- Add `title`, `date`, `author`, and `tags` ([view existing tags](https://www.gatsbyjs.org/blog/tags/) or add a new one) to the frontmatter of your `index.md`. If you are cross posting your post you can add `canonicalLink` for SEO benefits. You can check the other blog posts in `/docs/blog` for examples.
+- If your blog post contains images add them to your blog post folder and reference them in your post's `index.md`.
+- Ensure any links to gatsbyjs.org are relative links - `/contributing/how-to-contribute/` instead of `https://gatsbyjs.org/contributing/how-to-contribute`
+- Follow the [Style Guide](https://www.gatsbyjs.org/contributing/gatsby-style-guide/#word-choice) to make sure you're using the appropriate wording.
+- Double check your grammar and capitalise correctly
+- Commit and push to your fork
+- Create a pull request from your branch
+  - We recommend using a prefix of `docs`, e.g. `docs/your-change` or `docs-your-change`
```

Comment:
Should we link to a canonical example PR, such as https://github.com/gatsbyjs/gatsby/commit/9c21394add7906974dcfd22ad5dc1351a99d7ceb#diff-bf544fce773d8a5381f64c37d48d9f12?

Reply:
Yes! Great idea.